### PR TITLE
add linting and fix lintballs

### PR DESCRIPTION
**Pull request by michaelshobbs**
lint with `golint` and `go tool vet`

### CODE REVIEW

File: adapters/syslog/syslog.go [hunk 1]
```diff
@@ -112,7 +115,7 @@ func (a *SyslogAdapter) Stream(logstream chan *router.Message) {
 	}
 }
 
-func (a *SyslogAdapter) retry(buf []byte, err error) error {
+func (a *Adapter) retry(buf []byte, err error) error {
```
Comment (josegonzalez) [inline adapters/syslog/syslog.go]: Should internal functions also have a docblock?

Comment (michaelshobbs) [inline adapters/syslog/syslog.go]: It's not required or enforceable. However, if a function is doing something complex or "needs" additional explanation, then I tend to add them in that case.

**Comment by progrium** [comment]
Can you simplify the Makefile? Maybe less Bash scripting?

**Comment by michaelshobbs** [comment]
@progrium i played with this quite a bit. open to suggestions.

File: Makefile [hunk 2]
```diff
@@ -18,6 +22,13 @@ build:
 	docker build -t $(NAME):$(VERSION) .
 	docker save $(NAME):$(VERSION) | gzip -9 > build/$(NAME)_$(VERSION).tgz
 
+lint:
+	test -x $(GOPATH)/bin/golint || go get github.com/golang/lint/golint
+	go get \
+		&& go install \
+		&& ls -d */ | egrep -v '/custom/' | xargs $(XARGS_ARG) go tool vet -v
+	@if [ -n "$(shell $(GOLINT))" ]; then $(GOLINT) && exit 1; fi
```
Comment (josegonzalez) [inline Makefile]: `ifneq ($(GOLINT),)` will do.

Comment (michaelshobbs) [inline Makefile]: if blocks run on all targets. i don't think we want that.

Comment (josegonzalez) [inline Makefile]: Yeah but the var is already always set, no?

Comment (josegonzalez) [inline Makefile]: Oh nvm `:=`. Dumb me.

File: adapters/syslog/syslog.go (hunk 1 above)
Comment (josegonzalez) [inline adapters/syslog/syslog.go]: cool cool

File: Makefile (hunk 2 above)
Comment (michaelshobbs) [inline Makefile]: Yeah `GOLINT` is a string that only gets processed in the `lint` target

**Comment by josegonzalez** [comment]
We can simplify the makefile at a later date. Getting linting working is probably the first priority :)

**Comment by josegonzalez** [comment]
By we I mean @michaelshobbs (lord knows I'm not touching this).

**Pull request merged**